datalad/support/annexrepo.py

```diff
@@ -1169,6 +1169,20 @@ def supports_unlocked_pointers(self):
             # that unlocked pointers aren't supported.
             return False
 
+    @property
+    def auto_upgradeable_versions(self):
```

Comment:
I wonder if it is worth adding it to AnnexRepo's already extended public API, especially since it doesn't need `self` at all... May be just a little private helper, @classmethod if to be bound somehow?

Reply:
I'm fine either way.  It would just need to be a top-level dict in the module.  Assuming your wondering is actually a preference, I'll change it to that.

Reply:
Update pushed:

```
$ git range-diff kyleam/enh-log-repo-upgrade...enh-log-repo-upgrade
1:  0bf7fa7d4 < -:  --------- ENH: annexrepo: Add auto_upgradeable_versions property
2:  6a1e290d2 ! 1:  abed6efad ENH: annexrepo: Inform users about repo version auto-upgrades
    @@ -12,11 +12,25 @@
     diff --git a/datalad/support/annexrepo.py b/datalad/support/annexrepo.py
     --- a/datalad/support/annexrepo.py
     +++ b/datalad/support/annexrepo.py
    +@@
    + # Limit to # of CPUs and up to 8, but at least 3 to start with
    + N_AUTO_JOBS = min(8, max(3, cpu_count()))
    + 
    ++# This is a map between an auto-upgradeable version and the version that it
    ++# upgrades to. It should track autoUpgradeableVersions in Annex.Version.
    ++_AUTO_UPGRADEABLE_VERSIONS = {3: 5, 4: 5}
    ++# TODO: Adjust once GIT_ANNEX_VERSION is at least 7.20181031.
    ++if external_versions['cmd:annex'] >= '7.20181031':
    ++    _AUTO_UPGRADEABLE_VERSIONS[6] = 7
    ++
    + 
    + class AnnexRepo(GitRepo, RepoInterface):
    +     """Representation of an git-annex repository.
     @@
              if description is not None:
                  opts += [description]
              if version is not None:
    -+            upgraded_version = self.auto_upgradeable_versions.get(version)
    ++            upgraded_version = _AUTO_UPGRADEABLE_VERSIONS.get(version)
     +            if upgraded_version:
     +                lgr.info("Annex repository version %s will be upgraded to %s",
     +                         version, upgraded_version)
```